add flatpak manifest

dreamyukii opened the pull request:
add flatpak manifest for Kanji Dojo.

dreamyukii committed:
add flatpak manifest

syt0r commented on .vscode/settings.json:
```diff
@@ -0,0 +1,10 @@
+{
```
I believe this file should be ignored

syt0r commented on flatpak/io.github.syt0r.kanji_dojo.yml:
```diff
@@ -0,0 +1,43 @@
+          type: json
+          url: https://api.github.com/repos/syt0r/Kanji-Dojo/releases/latest
+          version-query: .tag_name | sub("^v"; "")
+          url-query: '"https://github.com/syt0r/Kanji-Dojo/releases/download/" + $version + "/kanji-dojo-linux-x64-" + $version ".jar"'
```
I actually removed the jar distributable in favor of AppImage a few commits ago so this will require some adjustments.  Here's the place where the AppImage is built with GitHub Action, I believe flatpak version needs something similar

https://github.com/syt0r/Kanji-Dojo/blob/a6b1162a0208bb254337a2e0758a327eb3c49cf4/.github/workflows/build-all.yml#L29-L31

https://github.com/syt0r/Kanji-Dojo/blob/a6b1162a0208bb254337a2e0758a327eb3c49cf4/AppImage/make.sh#L7

I suppose that the `gradlew :core:createDistributable` command creates a standalone app with a stripped jvm bundled in it, the output is written to `core/build/compose/binaries/main/app/Kanji Dojo` so we can additionally publish that folder or if there's a flatpak file format we can build that instead

dreamyukii replied:
I think publish that folder is fine, also how about just repackaged the Appimage format? lot of Flatpak apps on flathub do that.

syt0r replied:
That's great then. I'll create the next release with a small fix and this new AppImage file attached soon because seems like now it might be hard to implement any changes

syt0r reviewed (commented):
Thanks for your efforts! In addition to my other comments can you please list some commands that I can use to verify it locally?

dreamyukii commented:
> Thanks for your efforts! In addition to my other comments can you please list some commands that I can use to verify it locally?

you mean command to build flatpak locally?

dreamyukii committed:
remove .vscode

syt0r commented:
> > Thanks for your efforts! In addition to my other comments can you please list some commands that I can use to verify it locally?
> 
> you mean command to build flatpak locally?

Yes, like build/install/remove, if it's possible, I've never created flatpak apps before

dreamyukii commented:
> > > Thanks for your efforts! In addition to my other comments can you please list some commands that I can use to verify it locally?
> > 
> > 
> > you mean command to build flatpak locally?
> 
> Yes, like build/install/remove, if it's possible, I've never created flatpak apps before

First you need to[ install Flatpak](https://flatpak.org/setup/) if you don't have it installed in your system, then add flathub remote with --user flags, `flatpak remote-add --if-not-exists flathub https://dl.flathub.org/repo/flathub.flatpakrepo`. 

After that you also need to install flatpak builder, `flatpak install flathub  org.flatpak.Builder`.

Then you can run this command in flatpak folder to build the flatpak and install it `flatpak run org.flatpak.Builder --force-clean --sandbox --user --install --install-deps-from=flathub --repo=repo builddir io.github.syt0r.kanji_dojo.yml`.

You can also look at [flatpak docs](https://docs.flatpak.org/en/latest/building.html) for more information

dreamyukii committed:
switch to appimage for source

syt0r committed:
Enable sound, reuse default linux user data directory and fix links not opening in flatpak app

syt0r commented:
I've managed to verify it and added some fixes. One weird thing is that on my ubuntu the app icon looks blurry after installation, but looks like it might be a common bug?  
As I understand I'll have to create a pr to flathub with the .yml file from this pr?

<img src="https://github.com/user-attachments/assets/28d65d9f-45d4-4cee-be5c-fb39b4691e9a" width=500/>

dreamyukii committed:
change icon to 256x256

dreamyukii commented:
I think its because the icon file is 256x256 but i'm put in at 512x512 icons folder, also i already create branch for [flathub submission ](https://github.com/dreamyukii/flathub/tree/io.github.syt0r.kanji_dojo)

syt0r committed:
Move all linux app packaging data and scripts into a single folder
Replace underscore with dash in flatpak id
Add more flatpak screenshots
Update flatpak summary, description, branding

syt0r commented:
Your fix with the icon worked, thanks! In the last commit I've made some changed, please read the full commit message, is everything looks alright? Also I'm not sure whether there should be screenshot section in metainfo

dreamyukii commented:
Screenshot is a requirement for flathub submission

dreamyukii commented:
> Your fix with the icon worked, thanks! In the last commit I've made some changed, please read the full commit message, is everything looks alright? Also I'm not sure whether there should be screenshot section in metainfo

everything is ok!

syt0r committed:
Add more info to Flathub's metainfo

syt0r commented:
I've added some extra info to metadata, everything except for image urls is alright according to linter I run with a command `flatpak run --command=flatpak-builder-lint org.flatpak.Builder appstream io.github.syt0r.kanji-dojo.metainfo.xml`, urls should work after it's merged

dreamyukii commented:
> Validation is successful now, thanks! I guess you can try to submit it now with the updated data (note that the id was changed so your branch should probably be renamed). Or should I handle submitting?

i think its better for upstream to submit the submission. also don't forget to verify the apps after submitting
